scripts/generator-adapter/generators/app/index.ts

```diff
@@ -0,0 +1,441 @@
+import * as Generator from 'yeoman-generator'
+interface InputTransport {
+  type: string
+  name: string
+}
+interface EndpointGenerator {
+  // The name of endpoint from user input. Spaces replaced with '-'
+  inputEndpointName: string
+  // Endpoint name converted to normalized name that can be used in imports/exports, i.e. crypto-one-two -> cryptoOneTwo
+  normalizedEndpointName: string
+  // List of user-selected transports
+  inputTransports: InputTransport[],
+  // List of endpoint aliases
+  endpointAliases: string[]
+  // Endpoint name with a first letter being uppercase, i.e. crypto -> Crypto. This is needed for request payload names in the tests
+  normalizedEndpointNameCap: string
+}
+
+module.exports = class extends Generator {
+  props: {
+    // Current ea-framework version in package.json
+    frameworkVersion: string
+    adapterName: string,
+    endpoints: Record<string, EndpointGenerator>,
+    // Comma seperated list of endpoints
+    endpointNames: string,
+    defaultEndpoint: EndpointGenerator
+    // Whether to include explanation comments for different parts of adapter components
+    includeComments: boolean
+  }
+
+  constructor(args, opts) {
+    super(args, opts);
+    this.argument("rootPath", { type: String, required: false, default: './', description: 'Root path where new External Adapter will be created'});
+  }
+  // prompting stage is used to get input from the user, validate and store it to use it for next stages
+  async prompting() {
+    const adapterName = await this._promptAdapterName()
+    const endpointCount = await this._promptEndpointCount()
+    const endpoints: Record<string, EndpointGenerator> = {}
+
+    for (let i = 0; i < endpointCount; i++) {
+      let inputEndpointName = await this._promptEndpointName(i)
+      const existingEndpoint = this._checkExistingEndpoint(endpoints, inputEndpointName)
+
+      // If the user provides the same endpoint name for a different endpoint, try prompting again a few times to use another name.
+      if (existingEndpoint) {
+        let retries = 3
+        while (retries !== 0) {
+          inputEndpointName = await this._promptEndpointName(i, inputEndpointName)
+          const existingEndpointRetry = this._checkExistingEndpoint(endpoints, inputEndpointName)
+
+          if (!existingEndpointRetry) {
+            break;
+          }
+          --retries
+
+          if (retries === 0) {
+            throw new Error(`There is already configured endpoint with name '${inputEndpointName}'`)
+          }
+        }
+      }
```

Comment:
I think chunks like this could be simplified, as well as ignoring max retries (just let the user quit eventually)
```typescript
endpointNames = new Set()

  ...

  
  promptEndpointName(index: number) {
    const { inputEndpointName } = await this.prompt<{inputEndpointName: string}>({
      type: "input",
      name: "inputEndpointName",
      message: `What is the name of endpoint #${index+1}:`,
      default: "price"
    })

    const endpointName = this._normalizeStringInput(inputEndpointName)

    if (endpointName === '') {
      console.log('Endpoint name cannot be empty')
      this.promptEndpointName(index)
    }
    
    if (this.endpointNames.has(endpointName)) {
      console.log(`Endpoint name ${endpointName} already exists`)
      this.promptEndpointName(index)
    }

    return endpointName
  }
```

Reply:
Then we can do
```suggestion
    for (let i = 0; i < endpointCount; i++) {
      const endpointName = await this._promptEndpointName(i)
```

Reply:
And same for the aliases chunk below

Reply:
Hey @alejoberardino , is there any particular reason to use `Set` for endpoint+alias names ? Currently there are 
`_checkExistingEndpoint` and `_checkExistingEndpointAliases` that check if there is any endpoint or alias for given name. We could use Set just to make sure that aliases for the same endpoint don't contain duplicates but after that convert to array again since working with sets requires additional changes in the templates as well. 

so the same example you provided 
```typescript
if (this.endpointNames.has(endpointName)) {
      console.log(`Endpoint name ${endpointName} already exists`)
      this.promptEndpointName(index)
}
```
can be 
```typescript
const existingEndpoint = this._checkExistingEndpoint(endpoints, inputEndpointName)
    if (existingEndpoint) {
      this.log(`Endpoint named or aliased '${endpointName}' already exists`)
      return this._promptEndpointName(index, endpoints)
    }
```

Reply:
We can keep both structures, the set would be pretty minimal and make checks run in constant time (not a huge deal given the actual iterations, but would remove the loops from the code